File: greatest_common_divisor/greatest_common_divisor.py
def gcd(a: int, b: int) -> int:
    '''
    Greates common divisor of two integers(Euclidean algorithm)

    Parameters
    ----------
    a, b : int
        The algorithm will use abslute values
        Example: gcd(-3, 9) == gcd(3, 9)
    '''
    a, b = abs(a), abs(b)

    while b != 0:
        b, a = a % b, b
    return a


if __name__ == '__main__':
    print('GCD({}, {}) = {}'.format(9, 12, gcd(9, 12)))
    print('GCD({}, {}) = {}'.format(-3, 32, gcd(-3, 32)))
    print('GCD({}, {}) = {}'.format(14, 21, gcd(14, 21)))
    print('GCD({}, {}) = {}'.format(3, 0, gcd(3, 0)))
    print('GCD({}, {}) = {}'.format(0, 0, gcd(0, 0)))

File: greatest_common_divisor/greatest_common_divisor.c
#include <stdio.h>

// Returns greates common divisor of two number
unsigned int gcd(unsigned int a, unsigned int b){

    unsigned int tmp;
    while(b != 0){
        tmp = b;
        b = a % b;
        a = tmp; 
    }
    
    return a;
}

int main(){
    printf("GCD(%u, %u) = %u\n", 3, 9, gcd(3, 9));
    printf("GCD(%u, %u) = %u\n", 0, 4, gcd(0, 4));
    printf("GCD(%u, %u) = %u\n", 0, 0, gcd(0, 0));
    printf("GCD(%u, %u) = %u\n", 21, 56, gcd(21, 56));

    return 0;
}
